src/main/java/earl/tasks/Task.java

```diff
@@ -0,0 +1,61 @@
+package earl.tasks;
+
+/**
+ * Abstract class encapsulating a task to be tracked.
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Class constructor.
+     *
+     * @param description  a {@code String} naming the task
+     */
+    public Task(String description) {
+        this.description = description;
+        isDone = false;
+    }
+
+    /**
+     * Class constructor specifying the current completion status.
+     *
+     * @param status       a {@code String} that is either "X" or " "
+     * @param description  a {@code String} naming the task
+     */
+    public Task(String status, String description) {
+        this.description = description;
+        isDone = status.equals("X");
+    }
+
+    /** Returns a visual indicator of completion. */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /** Marks as complete. Returns success. */
+    public boolean markAsDone() {
+        if (!this.isDone) {
+            this.isDone = true;
+            return true;
+        }
+        return false;
+    }
+
+    /** Marks as incomplete. Returns success. */
+    public boolean markUndone() {
+        if (this.isDone) {
+            this.isDone = false;
+            return true;
+        }
+        return false;
+    }
```

Comment:
Perhaps the boolean methods can be renamed to sound like booleans, e.g. isMarkedDone, isMarkedUndone.

Reply:
The boolean here refers to the success of changing the completion status of the task; it returns false if it is already of the desired state. I can see that this part may not be clear what it is doing, I will think of a way to improve its clarity. Thanks!